src/RegExpr.pas

```diff
@@ -975,24 +986,49 @@ function ReplaceRegExpr(const ARegExpr, AInputStr, AReplaceStr: RegExprString;
 end;
 {$ENDIF}
 
+(*
 const
-  META: RegExprString = '^$.[()|?+*' + EscChar + '{';
+  MetaChars_Init = '^$.[()|?+*' + EscChar + '{';
+  MetaChars = MetaChars_Init; // not needed to be a variable, const is faster
+  MetaAll = MetaChars_Init + ']}'; // Very similar to MetaChars, but slighly changed.
+*)
+
+function _IsMetaSymbol1(ch: REChar): boolean; {$IFDEF InlineFuncs}inline;{$ENDIF}
+begin
+  case ch of
+    '^', '$', '.', '[', '(', ')', '|', '?', '+', '*', EscChar, '{':
+      Result := True
+    else
+      Result := False
+  end;
+end;
+
+function _IsMetaSymbol2(ch: REChar): boolean; {$IFDEF InlineFuncs}inline;{$ENDIF}
+begin
+  case ch of
+    '^', '$', '.', '[', '(', ')', '|', '?', '+', '*', EscChar, '{',
+    ']', '}':
+      Result := True
+    else
+      Result := False
+  end;
+end;
 
 function QuoteRegExprMetaChars(const AStr: RegExprString): RegExprString;
 var
-  MetaAll: RegExprString; // Very similar to META, but slighly changed.
   i, i0, Len: integer;
+  ch: REChar;
 begin
   Result := '';
-  MetaAll := META + ']}';
   Len := Length(AStr);
   i := 1;
   i0 := i;
   while i <= Len do
   begin
-    if Pos(AStr[i], MetaAll) > 0 then
+    ch := AStr[i];
```

Comment:
I am sure the compiler is smart enough to place it to CPU register so this change does improve readability but add nothing to performance.